src/test/java/com/example/demoSpringRestaurant/unit/service/RestaurantServiceTest.java

```diff
@@ -31,28 +35,92 @@ public class RestaurantServiceTest {
         when(restaurantMapper.fromEntityToRestaurantDto(any(RestaurantEntity.class)))
                 .thenReturn(RestaurantFixture.getRestaurantDto());
         when(restaurantRepository.save(any(RestaurantEntity.class)))
                 .thenReturn(RestaurantFixture.getRestaurantEntity(true));
+
         //act / when
         var restaurantDto = restaurantService.addRestaurant(RestaurantFixture.getRestaurantCreationDto());
 
         //assert / then
         assertThat(restaurantDto).usingRecursiveComparison().isEqualTo(RestaurantFixture.getRestaurantDto());
-        verify(restaurantRepository,times(1)).save(any(RestaurantEntity.class));
+        verify(restaurantRepository, times(1)).save(any(RestaurantEntity.class));
+        verifyNoMoreInteractions(restaurantRepository);
+    }
+
+    @Test
+    void getRestaurantShouldGetAllRestaurant() {
+        when(restaurantMapper.fromEntityToRestaurantDto(any(RestaurantEntity.class)))
+                .thenReturn(RestaurantFixture.getRestaurantDto());
+        when(restaurantRepository.findAll()).thenReturn(RestaurantFixture.getRestaurantEntityList());
+
+        //act / when
+        var restaurantDtoList = restaurantService.getRestaurants();
+
+        //assert / then
+        assertThat(restaurantDtoList).usingRecursiveComparison().isEqualTo(RestaurantFixture.getRestaurantDtoList());
+        verify(restaurantRepository, times(1)).findAll();
+        verifyNoMoreInteractions(restaurantRepository);
+    }
+
+    @Test
+    void updateRestaurantShouldUpdateOneRestaurant() throws EntityNotFoundException {
+        //arrange / given
+        when(restaurantMapper.fromEntityToRestaurantDto(any(RestaurantEntity.class)))
+                .thenReturn(RestaurantFixture.getRestaurantDto());
+        when(restaurantMapper.fromRestaurantUpdateDtoToEntity(any(RestaurantUpdateDto.class)))
+                .thenReturn(RestaurantFixture.getRestaurantEntity(false));
+        when(restaurantRepository.save(any(RestaurantEntity.class)))
+                .thenReturn(RestaurantFixture.getRestaurantEntity(true));
+        when(restaurantRepository.existsById(any(Long.class))).thenReturn(true);
+        //act / when
+        var restaurantUpdate = restaurantService.updateRestaurant(5L, RestaurantFixture.getRestaurantUpdateDto());
+        var restaurantUpdate2 = restaurantService.updateRestaurant(1L, RestaurantFixture.getRestaurantUpdateDto());
+
+        //assert / then
+        assertThat(restaurantUpdate.getId()).isEqualTo(1L);
+        assertThat(restaurantUpdate2.getId()).isEqualTo(1L);
+        assertThat(restaurantUpdate).usingRecursiveComparison().isEqualTo(RestaurantFixture.getRestaurantDto());
+        assertThat(restaurantUpdate).isEqualTo(restaurantUpdate2);
+        verify(restaurantRepository, times(2)).existsById(any(Long.class));
+        verify(restaurantRepository, times(2)).save(any(RestaurantEntity.class));
+        verifyNoMoreInteractions(restaurantRepository);
```

Comment:
Itt a két különböző update hívásra miért van szükség?

Az első két assert kicsit fura, főleg, hogy a `restaurantUpdate`-nek 5 az ID-ja, utána pedig azt validáljuk, hogy 1 lett-e. Ez nyilván a mock miatt van így, ami nem is baj, csak így meg elég lenne egy eset, ahol látszik, hogy minek kell hívódnia repository oldalon.

Illetve itt is jó lenne plusz tesztet hozzáadni a hibakezelésre, pl ha nem létezik az adott étterem.